### PR TITLE
Add Callbacks portion to FFI API manual

### PR DESCRIPTION
**Pull request by aapoalas**
Relates to https://github.com/denoland/deno/pull/14663

### CODE REVIEW

File: runtime/ffi_api.md [hunk 1]
```diff
@@ -134,28 +134,118 @@ Before
...
+| `f32`         | `float`                  | `f32`       |
+| `f64`         | `double`                 | `f64`       |
+| `void`        | `void`                   | `()`        |
+| `pointer`[1]  | `const uint8_t *`        | `*const u8` |
+| `function`[2] | `void (*fun)()`          | `fn()`      |
 
 - [1] `pointer` type accepts both Typed Arrays and `Deno.UnsafePointer` as
   parameter, while it always returns the latter when used as result type.
+- [2] `function` type is defined as an object containing its parameters and
+  return value, which can be any of the supported types.
```
[the remark this answers is not in the document]
Comment (aapoalas) [reply runtime/ffi_api.md]: I didn't change this too much as this was about the definition, not the parameter / result types.

File: runtime/ffi_api.md [hunk 2]
```diff
@@ -134,28 +134,118 @@ Before
...
+If an interrupt handler calls back into Deno, one of a few things may happen:
+
+1. If Deno is currently idle, the callback will be called on an empty stack
+   successfully. Any errors thrown by the callback cannot be caught.
+2. If Deno is presently running code, the callback will be run on that stack
+   when and where V8 happens to yield the thread. Any errors thrown will
+   propagate into the presently running code, causing very much unexpected
+   errors to be thrown.
+3. IF Deno is presently running code that calls appropriate functions, the
+   program will instantly exit with no warning.
```
[the remark this answers is not in the document]
Comment (aapoalas) [reply runtime/ffi_api.md]: Removed.

File: runtime/ffi_api.md [hunk 3]
```diff
@@ -134,28 +134,118 @@ Before
...
+
+// After callback is no longer needed
+callback.close();
+// It is now safe to let the callback go out of scope. It is no longer safe
+// to pass the callback as a parameter.
+```
+
+If the `close()` method is called while the C function pointer is still held by
+a dynamic library, then calling that function pointer becomes undefined
+behaviour and will likely crash the program.
```
[the remark this answers is not in the document]
Comment (aapoalas) [reply runtime/ffi_api.md]: Removed.

**Review by littledivy — APPROVED** [review]
LGTM!

**Pull request merged**